lightning/src/util/ser.rs

```diff
@@ -803,6 +803,39 @@ impl_for_vec!(ecdsa::Signature);
 impl_for_vec!(crate::ln::channelmanager::MonitorUpdateCompletionAction);
 impl_for_vec!((A, B), A, B);
 
+impl Writeable for Vec<Witness> {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		(self.len() as u16).write(w)?;
```

Comment:
I agree with consensus serialization here. It didn't really fly when I brought it up on spec.

Reply:
Happy to do this, and will let y'all pre-review.

Reply:
So from what I understand is that we just want to propose the [compactsize](https://en.bitcoin.it/wiki/Protocol_documentation#Variable_length_integer) fundamental type in a separate PR and motivate its usage where bitcoin-consensus serialization is desired. Then basically just use that to define the witness lists and other future things in the places in the spec they're used. 

Is an appendix necessary with the test vectors like with BigSize since we're hoping that an upstream lib is used by implementations, or do we just go ahead and provide its full definition in the spec with test vectors and all?

Reply:
So like get this in as a separate PR and then propose the changes to the witnesses encoding in the dual funding PR (I believe Wilmer mentioned adding compactsize as a new fundamental type before).

https://github.com/dunxen/bolts/commit/b4f52f3bea498f0ae6daaa82dd9e21ff3c5dc48b

I mostly just copied the existing BigSize stuff, reversed endianness, and tested against rust-bitcoin locally (kept the golang example lol). Also wasn't sure whether to modify the appendix letters or just make it the next one, namely, D. I should probably make it explicit that it's little-endian in the description before the test vectors?

Reply:
Have the changes to `tx_signatures` also up (dependent on https://github.com/lightning/bolts/pull/1062)
https://github.com/niftynei/lightning-rfc/pull/14